### PR TITLE
feat: added Odin language server yaml

### PR DESCRIPTION
**Pull request by WillEhrendreich**
Hey @williamboman, @DanielGavin,and @gingerbill!
I thought it would be great for this to be an automatic download for neovim users, like so many other language servers are, so this is my attempt at helping solve that. If I didn't do something right, then please feel free to change it. I've never contributed to a mason registry before, so I hope this is in the right direction.

### CODE REVIEW

**Comment by williambotman** [comment]
@mason-org/triage `notify(Reason::NewPullRequest)`

<sup>This comment ensures a notification is delivered to maintainers.</sup>

<sup>` 🤖 This is an automated comment. `  [` 📖 Source code `](https://github.com/williamboman/botman)</sup>

File: packages/ols/ols.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - MIT
+languages:
+  - Odin
+categories:
+  - LSP
+
+source:
+  id: pkg:github/DanielGavin/ols@nightly
+  asset:
+    - target: darwin_x64
```
Comment (williamboman) [inline packages/ols/ols.yaml:15]: This binary is executable on Apple chips as well via [Rosetta](https://en.wikipedia.org/wiki/Rosetta_(software))

```suggestion
    - target: [darwin_x64, darwin_arm64]
```

**Review by williamboman — APPROVED** [review]
LGTM!

File: packages/ols/ols.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  asset:
+    - target: [darwin_x64, darwin_arm64]
+      file: ols-x86_64-darwin
+      bin: ols-x86_64-darwin
+    - target: linux_x64
+      file: ols-x86_64-unknown-linux-gnu 
+      bin: ols-x86_64-unknown-linux-gnu
+    - target: linux_x86
+      file: ols-x86_64-unknown-linux-gnu 
+      bin: ols-x86_64-unknown-linux-gnu
```
Comment (williamboman) [inline packages/ols/ols.yaml:23]: Or wait, these are GNU binaries for x64 only

```suggestion
    - target: linux_x64_gnu
      file: ols-x86_64-unknown-linux-gnu 
      bin: ols-x86_64-unknown-linux-gnu
```

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/packages/ols/ols.yaml b/packages/ols/package.yaml
similarity index 100%
rename from packages/ols/ols.yaml
rename to packages/ols/package.yaml
```

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/schemas/enums/language.json b/schemas/enums/language.json
index 16a3d18..a52ab2f 100644
--- a/schemas/enums/language.json
+++ b/schemas/enums/language.json
@@ -93,6 +93,7 @@
         "Nix",
         "Nunjucks",
         "OCaml",
+        "Odin",
         "OneScript",
         "OpenAPI",
         "OpenCL",
```

**Pull request merged**